resources/glossary/graphs.mdx

```diff
@@ -0,0 +1,27 @@
+---
+objectId: 5g7f9g17-0205-4043-f46f-0ba3793i4616
+title: Graphs
+description: "A graph is a collection of entities which may be connected to other entities by links."
+slug: graphs
+tags: ["Graphs", "Machine Learning", "Software Engineering"]
+---
+
+## What are graphs?
+
+### In HASH
+
+In the context of HASH, a graph is a collection of entities which may be connected to other entities by [links](https://hash.ai/glossary/links).
+
+The data in a single HASH workspace can be thought of as a graph, some of which may be kept private, and parts of which may be public.
+
+The totality of all public data in HASH may also be thought of as a graph, as many individual workspaces are connected through [linked entities](https://hash.ai/glossary/entities#Linked-entities) and shared [types](https://hash.ai/glossary/types).
+
+### In the Block Protocol
+
+The [Block Protocol](https://hash.ai/glossary/block-protocol) service which deals with entities and the links between them is known as the graph service.
+
+### In mathematics and computer science
+
+A graph is a data structure with **nodes** and **edges**. Nodes are sometimes also called ‘vertices’ or ‘points’, and edges are sometimes also referred to as ‘lines’ or ‘links’ (the term we use in HASH).
+
+The edges of graphs can be **directed** or **undirected**, reflecting the nature of a connection between two nodes. For example, an edge reflecting a ‘sibling’ relationship between two sisters would be undirected, but an edge representing a debt owed by one to the other would be directed.
```

Comment:
Just to flag, in the current design due to compromises, all links are directed. So to represent a bi-directional relationship there would be a sibling link for each direction rather than one bidirectional one.